core/src/main/java/org/apache/mahout/common/distance/EuclideanDistanceMeasure.java

```diff
@@ -29,26 +31,28 @@
  * squared differences.
  */
 public class EuclideanDistanceMeasure extends SquaredEuclideanDistanceMeasure {
-  
+  Logger log = LoggerFactory.getLogger(EuclideanDistanceMeasure.class);
   @Override
   public double distance(Vector v1, Vector v2) {
     double largeDistanceValue = 1000000;
-    int [] fieldNumbersThatMustMatch = {3}; // Field 3 is dwellingType
+    int [] fieldNumbersThatMustMatch = {6}; // Field 3 is dwellingType
```

Comment:
This code that generates the weighted vector data tells me it is 3 and not 6 .

        for (int i = 3; i < accountAndModifiedTimeAndFeatures.length; i++) {
            try {
                Double val = Double.parseDouble(accountAndModifiedTimeAndFeatures[i]);
                features[i - 3] = val;
            }
            catch (NumberFormatException nfe) {
                throw new TendrilException("String '" + accountAndModifiedTimeAndFeatures[i] + "' in position "
                        + (i) + " could not be parsed. The full string is '" +
                        line + "'", nfe);
            }

        }

        NamedVector weightedVector = createWeightedVector(accountId, features);


The data:

vector_data =
	FOREACH
		accounts_and_bills_and_demographics
	GENERATE
		account_data::id as account_id,
		account_data::last_modified as account_data_last_modified,
		demographics::last_modified as demographics_last_modified,
		demographics::location_element_1 as location_element_1,
        demographics::location_element_2 as location_element_2,
        demographics::location_element_3 as location_element_3,
		DwellingTypeCodeSimplificationUdf(demographics::dwelling_type_code) as dwelling_type_scalar,
		demographics::square_footage as square_footage,
		ISO_DAYS_BETWEEN(
			latest_bills::bill_cycle_end,
			latest_bills::bill_cycle_start
		) as bill_cycle_length_days,
		demographics::year_built as year_built,
        HeatingTypeCodeSimplificationUdf(demographics::heating_type_code) as heating_type_scalar,
        StateCodeSimplificationUdf(account_data::location_state) as state_code_scalar;